pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -130,6 +134,51 @@ func (rc *remoteClient) queueWorkloadEvent(ctx context.Context, ev watch.Event)
 	}
 }
 
+// runGc - lists all the remote workloads having the same multikueue-origin and remove those who
+// no longer have a local corespondent (missing or awaiting deletion). If the remote workload is
+// is owned by a job,also delete the job.
+func (rc *remoteClient) runGC(ctx context.Context) {
+	log := ctrl.LoggerFrom(ctx)
+	lst := &kueue.WorkloadList{}
+	err := rc.client.List(ctx, lst, client.MatchingLabels{MultiKueueOriginLabelKey: rc.origin})
+	if err != nil {
+		log.V(2).Error(err, "Listing remote workloads")
+		return
+	}
+
+	for _, remoteWl := range lst.Items {
+		localWl := &kueue.Workload{}
+		wlLog := log.WithValues("remoteWl", klog.KObj(&remoteWl))
+		err := rc.localClient.Get(ctx, client.ObjectKeyFromObject(&remoteWl), localWl)
+		if client.IgnoreNotFound(err) != nil {
+			wlLog.V(2).Error(err, "Reading local workload")
+			continue
+		}
+
+		if err == nil && localWl.DeletionTimestamp.IsZero() {
+			// The remote workload is still relevant.
+			continue
+		}
+
+		// if the remote wl has a controller(owning Job), delete the job
```

Comment:
I think it could be useful for debugging if we log that deletion is being attempted. This is what the k8s GC does: https://github.com/kubernetes/kubernetes/blob/23728378380ec853676c4d5bf7b39ae8dbc7a74f/pkg/controller/podgc/gc_controller.go#L344.

Reply:
done